astroquery/mast/core.py

```diff
@@ -1,299 +1,46 @@
-
-    if not all(x in json_obj.keys() for x in ['info', 'data']):
-        raise KeyError("Missing required key(s) 'data' and/or 'info.'")
-
-    # determine database type key in case missing
-    type_key = 'type' if json_obj['info'][0].get('type') else 'db_type'
-
-    # for each item in info, store the type and column name
-    for idx, col, col_type, ignore_value in \
-            [(idx, x['name'], x[type_key], "NULL") for idx, x in enumerate(json_obj['info'])]:
-            # [(idx, x['name'], x[type_key], x['default_value']) for idx, x in enumerate(json_obj['info'])]:
-
-        # if default value is NULL, set ignore value to None
-        if ignore_value == "NULL":
-            ignore_value = None
-        # making type adjustments
-        if col_type == "char" or col_type == "STRING":
-            col_type = "str"
-            ignore_value = "" if (ignore_value is None) else ignore_value
-        elif col_type == "boolean" or col_type == "BINARY":
-            col_type = "bool"
-        elif col_type == "unsignedByte":
-            col_type = np.ubyte
-        elif col_type == "int" or col_type == "short" or col_type == "long" or col_type == "NUMBER":
-            # int arrays do not admit Non/nan vals
-            col_type = np.int64
-            ignore_value = -999 if (ignore_value is None) else ignore_value
-        elif col_type == "double" or col_type == "float" or col_type == "DECIMAL":
-            # int arrays do not admit Non/nan vals
-            col_type = np.float64
-            ignore_value = -999 if (ignore_value is None) else ignore_value
-        elif col_type == "DATETIME":
-            col_type = "str"
-            ignore_value = "" if (ignore_value is None) else ignore_value
 
-        # Make the column list (don't assign final type yet or there will be errors)
-        # Step through data array of values
-        col_data = np.array([x[idx] for x in json_obj['data']], dtype=object)
-        if ignore_value is not None:
-            col_data[np.where(np.equal(col_data, None))] = ignore_value
-
-        # no consistant way to make the mask because np.equal fails on ''
-        # and array == value fails with None
-        if col_type == 'str':
-            col_mask = (col_data == ignore_value)
-        else:
-            col_mask = np.equal(col_data, ignore_value)
-
-        # add the column
-        data_table.add_column(MaskedColumn(col_data.astype(col_type), name=col, mask=col_mask))
-
-    return data_table
+from . import utils
+from .auth import MastAuth
+from .cloud import CloudAccess
+from .discovery_portal import PortalAPI
+from .services import ServiceAPI
 
 
-@async_to_sync
-class MastClass(QueryWithLogin):
+class MastQueryWithLogin(QueryWithLogin):
     """
-    MAST query class.
-
-    Class that allows direct programatic access to the MAST Portal,
-    more flexible but less user friendly than `ObservationsClass`.
+    Super class for MAST functionality (should not be called directly by users).
```

Comment:
certainly add an `__all__` to this file, so users won't see the class even when they import core

Reply:
Oooooooh! I was trying to figure out how to do that! Excellent.